BuildTransportPackage.ps1

```diff
@@ -33,210 +37,257 @@ if(-not (test-path ".nuget\nuget.exe"))
+        
+        # If $AzureBuildStep is not "all", that means we are in the pipeline
+        $WindowsAppSDKBuildPipeline = 1
+
+        #------------------
+        #    Build windowsAppRuntime.sln and move output to staging.
+        #------------------
+        .\tools\GenerateDynamicDependencyOverrides.ps1 -Path "$buildOverridePath"
+        .\tools\GeneratePushNotificationsOverrides.ps1 -Path "$buildOverridePath"
+    }
+    if (($AzureBuildStep -eq "all") -Or (($AzureBuildStep -eq "BuildBinaries") -Or ($AzureBuildStep -eq "BuildMRT"))) 
+    {
+        & .\.nuget\nuget.exe restore WindowsAppRuntime.sln -configfile nuget.config
+        & .\.nuget\nuget.exe restore "dev\Bootstrap\CS\Microsoft.WindowsAppRuntime.Bootstrap.Net\Microsoft.WindowsAppRuntime.Bootstrap.Net.csproj" -configfile nuget.config
 
-$srcPath = Get-Childitem -Path 'dev\WindowsAppRuntime_Insights\packages' -File 'MicrosoftTelemetry.h' -Recurse
+        # If the call to restore WindowsAppRuntime_Insights fails check to make sure all Window SDK's from 17760 are installed.
+        & .\.nuget\nuget.exe restore "dev\WindowsAppRuntime_Insights\packages.config" -ConfigFile "dev\WindowsAppRuntime_Insights\nuget.config" -PackagesDirectory "dev\WindowsAppRuntime_Insights\packages"
 
-if (($srcPath -ne $null)){
-  $destinationPaths = Get-Childitem -Path 'packages' -File 'Traceloggingconfig.h' -Recurse
+        $srcPath = Get-Childitem -Path 'dev\WindowsAppRuntime_Insights\packages' -File 'MicrosoftTelemetry.h' -Recurse
 
-  if (($destinationPaths -ne $null)) {
-    foreach ($destPath in $destinationPaths) {
-      Write-Host 'SourcePath:' $srcPath.FullName
-      Write-Host 'DestinationPath:' $destPath.FullName
-      Copy-Item -Force $srcPath.FullName $destPath.FullName
-    }
-  }
-}
+        if (($srcPath -ne $null)){
+            $destinationPaths = Get-Childitem -Path 'packages' -File 'Traceloggingconfig.h' -Recurse
 
-foreach($configurationToRun in $configuration.Split(","))
-{
-    foreach($platformToRun in $platform.Split(","))
+            if (($destinationPaths -ne $null)) {
+                foreach ($destPath in $destinationPaths) {
+                Write-Host 'SourcePath:' $srcPath.FullName
+                Write-Host 'DestinationPath:' $destPath.FullName
+                Copy-Item -Force $srcPath.FullName $destPath.FullName
+                }
+            }
+        }
+    }
+    if (($AzureBuildStep -eq "all") -Or ($AzureBuildStep -eq "BuildBinaries")) 
     {
-        write-host "Building WindowsAppRuntime.sln for configuration $configurationToRun and platform:$platformToRun"
-        & $msBuildPath /restore WindowsAppRuntime.sln /p:Configuration=$configurationToRun,Platform=$platformToRun
+        foreach($configurationToRun in $configuration.Split(","))
+        {
+            foreach($platformToRun in $platform.Split(","))
+            {
+                write-host "Building WindowsAppRuntime.sln for configuration $configurationToRun and platform:$platformToRun"
```

Comment:
The YAML used the /m switch letting the build run multi-threaded.  Locally the build runs into a cyclical dependency issue when building with the /m switch.  If this issue can be figured out the /, can be re-added.

Reply:
Ah that might explain why the old pipelines were faster. I'll workout a solution.